spec/s3.spec.js

```diff
@@ -0,0 +1,192 @@
+"use strict";
+
+describe("storage/s3", () => {
+    var AwsSdkMock, promiseMock, s3;
+    
+    /**
+     * Checks the result for the promise object passed in.
+     *
+     * @param {Object} results
+     * @param {boolean} successful
+     * @param {string} message
+     */
+    function checkPromiseResults(results, successful, message) {
+        expect(results.success).toBe(successful);
+        expect(results.value).toEqual(message);
+    }
+    
+    beforeEach(() => {
+        var s3File;
+        
+        s3File = require("../lib/storage/s3");
+        promiseMock = require("./mock/promise-mock");
+        class S3Fake {
+            constructor(params) {
+                this.params = params;
+                this.deleteObjectAsync = jasmine.createSpy("deleteObjectAsync");
+                this.getObjectAsync = jasmine.createSpy("getObjectAsync");
+                this.listObjectsAsync = jasmine.createSpy("listObjectsAsync");
+                this.putObjectAsync = jasmine.createSpy("putObjectAsync");
+            }
+        }
+        AwsSdkMock = {
+            "S3": S3Fake,
+            "config": {
+                "region": null
+            }
+        };
+        s3 = new s3File(AwsSdkMock, promiseMock);
+    });
+    describe("configure", () => {
+        beforeEach(() => {
+            AwsSdkMock.S3 = jasmine.createSpy("s3Mock");
+        });
+        it("passes in configuration options for all", () => {
+            s3.transit();
+            expect(s3.aws.config.region).toBe("us-east-1");
+            expect(AwsSdkMock.S3).toHaveBeenCalledWith({params: {
+                Bucket: null
+            }});
+        });
+        it("passes in configuration options for all", () => {
+            s3.configure({
+                region: "us-west-1",
+                bucket: "test-bucket"
+            });
+            s3.transit();
+            expect(s3.aws.config.region).toBe("us-west-1");
+            expect(AwsSdkMock.S3).toHaveBeenCalledWith({params: {
```

Comment:
Object spacing again.  I could recommend a useful tool if it helps.  Could you go through the file and look for /. (slash followed by any character) and fix that for me?